### PR TITLE
New listsort.pl script: order /list by chan size

### PR DESCRIPTION
**Pull request by IsaacG**
New script which sorts the results of /list by channel size.

### CODE REVIEW

File: scripts/listsort.pl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            $list{$name}{'size'},
+            $name,
+            $list{$name}{'desc'},
+            $mode
+        );
+
+        Irssi::print($msg, MSGLEVEL_CRAP);
+    }
+    # Drop the hash values; no point in holding them in memory.
+    delete @list{keys %list};
```
Comment (ailin-nemui) [inline scripts/listsort.pl]: Use %list=() instead of this inefficient approach 😉

**Comment by IsaacG** [comment]
That's correct. Losing support for parallel /list's is something I think it worth it. In theory, I could track per-network lists and print them independently. However, the behaviour for parallel lists without this script isn't going to be super pretty, either. I'm not sure how servers would handle multiple /list commands on the same server; do they get serialized? If not, there's no solution for that.

File: scripts/listsort.pl (hunk 1 above)
Comment (IsaacG) [inline scripts/listsort.pl]: Done

**Comment by ailin-nemui** [comment]
yes, I meant for multiple servers. I guess it's not so frequently happening in real usage

**Comment by IsaacG** [comment]
Yup. I could handle list entries on a per-server basis but I'm not sure it's worth the additional code and room for bugs. People typically don't request a second list from another server before the first completes.

**Pull request merged**